File: Algorithms/src/153. Find Minimum in Rotated Sorted Array/Code/solution.go
func findMin(nums []int) int {
    
    // Left pointer
    left := 0

    // Right pointer
    right := len(nums) - 1

    // Binary Search loop
    for left < right {

        // Middle index
        mid := left + (right-left)/2

        // Minimum is on right side
        if nums[mid] > nums[right] {

            // Move left pointer
            left = mid + 1
        } else {

            // Minimum may be at mid
            right = mid
        }
    }

    // Return minimum element
    return nums[left]
}

File: Algorithms/src/153. Find Minimum in Rotated Sorted Array/Code/solution.java
class Solution {
    public int findMin(int[] nums) {

        // Left pointer at start
        int left = 0;

        // Right pointer at end
        int right = nums.length - 1;

        // Binary Search loop
        while (left < right) {

            // Find middle index
            int mid = left + (right - left) / 2;

            // Minimum is in right half
            if (nums[mid] > nums[right]) {

                // Move left pointer
                left = mid + 1;
            } else {

                // Minimum can be mid itself
                right = mid;
            }
        }

        // Return minimum element
        return nums[left];
    }
}